From f18e95849be93b86ff014147086fa538993abc21 Mon Sep 17 00:00:00 2001
From: Daisuke Kameda <daisuke@kde.gr.jp>
Date: Thu, 11 Mar 2010 13:39:15 +0100
Subject: [PATCH] Fixed problem displaying half width character as full width

There is the problem that Qt displays half width character
(e.g. alphabet) as full width. This is reported in #QTBUG-1726.

This occurs in the following fonts.
 - Monospace font
 - Selif font
 - IPA font (Ver.0301)
 - NSimSun

In Cario and fontconfig, the FT_LOAD_IGNORE_GLOBAL_ADVANCE_WIDTH
is specified as flag of FT_Load_Glyph function to solve this problem.
According to the document of freetype, this flag exists to support
buggy CJK fonts.

So this flag is specified as default flag of QFontEngineFT.

Merge-request: 2319
Reviewed-by: Eskil Abrahamsen Blomfeldt <eskil.abrahamsen-blomfeldt@nokia.com>
---
 src/gui/text/qfontengine_ft.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

--- a/src/gui/text/qfontengine_ft.cpp
+++ b/src/gui/text/qfontengine_ft.cpp
@@ -619,7 +619,7 @@ QFontEngineFT::QFontEngineFT(const QFont
     transform = false;
     antialias = true;
     freetype = 0;
-    default_load_flags = 0;
+    default_load_flags = FT_LOAD_IGNORE_GLOBAL_ADVANCE_WIDTH;
     default_hint_style = HintNone;
     subpixelType = Subpixel_None;
     lcdFilterType = 0;
